File: frontend/src/services/resumeService.js
import api from './api';

/**
 * @author Satheesh Kumar P
 * @since 2026-07-27
 * @version 1.0.0
 * 
 * @description API service module for resume download tracking and statistics.
 */
export const resumeService = {
  /**
   * Tracks a resume download event on the backend.
   * @returns {Promise<Object>} Response status
   */
  trackDownload: async () => {
    const response = await api.post('/resume/download');
    return response.data;
  },

  /**
   * Fetches resume download analytics.
   * @returns {Promise<Object>} { totalDownloads, last30DaysDownloads }
   */
  getStats: async () => {
    const response = await api.get('/resume/stats');
    return response.data;
  },
};

File: portfolio-service/src/main/java/com/satheesh/portfolio/service/ResumeService.java
package com.satheesh.portfolio.service;

import java.util.Map;

/**
 * @author Satheesh Kumar P
 * @since 2026-07-27
 * @version 1.0.0
 * 
 * @description Service interface for tracking and reporting resume download analytics.
 */
public interface ResumeService {

    /**
     * Records a resume download event with client metadata.
     * 
     * @param ipAddress Client IP address
     * @param userAgent Client User-Agent string
     * @param referer Client Referer URL
     */
    void trackDownload(String ipAddress, String userAgent, String referer);

    /**
     * Retrieves overall resume download statistics.
     * 
     * @return Map containing total count and 30-day count
     */
    Map<String, Object> getDownloadStats();
}
